src/main/java/opendcs/dao/TaskListDao.java

```diff
@@ -0,0 +1,117 @@
+package opendcs.dao;
+
+import decodes.tsdb.DbIoException;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Collection;
+
+
+import opendcs.dai.TaskListDAI;
+
+import org.opendcs.tsdb.FailedTaskListEntry;
+import org.opendcs.tsdb.TaskListEntry;
+
+public abstract class TaskListDao extends DaoBase implements TaskListDAI
+{
+    // Oracle was providing things in the wrong timestamp using current_timestamp.
+    // TODO: needs to be checked against Postgres
+    final String curTimeSqlCommand;
+    final String intervalSqlCommand;
+    public TaskListDao(DatabaseConnectionOwner dco)
+    {
+        this(dco, "tasklist");
+    }
+
+    public TaskListDao(DatabaseConnectionOwner dco, String module)
+    {
+        this(dco, module, null);
+    }
+
+    public TaskListDao(DatabaseConnectionOwner dco, String module, Connection con)
+    {
+        super(dco, module, con);
+        curTimeSqlCommand = dco.isOracle() ? "sysdate" : "current_timestamp" ;
+        intervalSqlCommand = dco.isOracle() ? " ?/24 )" : " ? * INTERVAL '1' hour )" ; // Oracle date math
```

Comment:
regarding DatabaseConnectionOwner.java 
 
As the code continues to refactor, Hopefully methods like : sOracle(), isHdb(), isCwms(), isOpenTSDB() could be eliminated, and instead each implementation can answer more specific questions, as needed, such as   getCurrentTimeStamp() .     

Another note: The name 'DatabaseConnectionOwner' feels like it should have the Connection if it is the owner, so why do we need a separate Connection argument.

Reply:
Yeah, that's the goal. At the moment I'm more focus on at least centralizing those things so that's easier to actually do.

As for DatabaseConnectionOwner, that should really be called DatabaseConnectionProvider but I'm fairly certain long term it's going to go away in favor of just "Database" so I'm not sure it's worth the effort to rename.